includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -703,32 +703,31 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 
 				// Create a payment intent, or update an existing one associated with the order.
 				$payment_intent = $this->process_payment_intent_for_order( $order, $payment_information );
-
-				// Handle saving the payment method in the store.
-				// It's already attached to the Stripe customer at this point.
-				if ( $payment_information['save_payment_method_to_store'] ) {
-					$this->handle_saving_payment_method(
-						$order,
-						$payment_information['payment_method'],
-						$selected_payment_type
-					);
-				}
-
-				// Use the last charge within the intent to proceed.
-				$charge = end( $payment_intent->charges->data );
-
-				// Only process the response if it contains a charge object. Intents with no charge require further action like 3DS and will be processed later.
-				if ( $charge ) {
-					$this->process_response( $charge, $order );
-				}
-
-				// Set the selected UPE payment method type title in the WC order.
-				$this->set_payment_method_title_for_order( $order, $selected_payment_type );
 			} else {
-				// It's a setup intent. To be handled.
-				return [ 'result' => 'failure' ];
+				$payment_intent = $this->process_setup_intent_for_order( $order, $payment_information );
+			}
+
+			// Handle saving the payment method in the store.
+			// It's already attached to the Stripe customer at this point.
+			if ( $payment_information['save_payment_method_to_store'] ) {
+				$this->handle_saving_payment_method(
+					$order,
+					$payment_information['payment_method'],
+					$selected_payment_type
+				);
+			} else if ( $payment_information['is_using_saved_payment_method'] ) {
```

Comment:
```suggestion
			} elseif ( $payment_information['is_using_saved_payment_method'] ) {
```
Nit: [Use elseif, not else if](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#use-elseif-not-else-if)